cmake/HPX_SetupTarget.cmake

```diff
@@ -174,7 +174,8 @@ function(hpx_setup_target target)
   if(NOT target_NOLIBS)
     target_link_libraries(
       ${target} ${__tll_public} $<TARGET_NAME_IF_EXISTS:hpx>
-      $<TARGET_NAME_IF_EXISTS:HPX::hpx>
+      $<TARGET_NAME_IF_EXISTS:HPX::hpx> $<TARGET_NAME_IF_EXISTS:wrap_main>
+      $<TARGET_NAME_IF_EXISTS:HPX::wrap_main>
```

Comment:
Let's put linking the `wrap_main` targets behind `if("{_type}" STREQUAL "EXECUTABLE")`.